pip_audit/_cli.py

```diff
@@ -239,6 +239,12 @@ def _parser() -> argparse.ArgumentParser:
         action="store_true",
         help="automatically upgrade dependencies with known vulnerabilities",
     )
+    parser.add_argument(
+        "--require-hashes",
+        action="store_true",
+        help="require a hash to check each requirement against, for repeatable audits; this option "
+        "is implied when any package in a requirements file has a --hash option.",
```

Comment:
```suggestion
        "is implied when any package in a requirements file has a `--hash` option.",
```

Reply:
We should also exit with an error if someone tries to use `--require-hashes` **without** `-r`. An argument group is probably the right approach for that.

Reply:
[f667aea](https://github.com/trailofbits/pip-audit/pull/229/commits/f667aea6939823e9c9b5a741d9642b9bd245cc15)

I don't believe I can use an argument group to say that an argument requires something else. The `-r` flag is also part of a mutually exclusive group with `--local`.

Instead I've just written a check that invokes `parser.error` if we use `--require-hashes` without `-r`.